oshi-core/src/main/java/oshi/software/os/mac/MacNetworkParams.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Oshi (https://github.com/dblock/oshi)
+ *
+ * Copyright (c) 2010 - 2017 The Oshi Project Team
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Maintainers:
+ * dblock[at]dblock[dot]org
+ * widdis[at]gmail[dot]com
+ * enrico.bianchi[at]gmail[dot]com
+ *
+ * Contributors:
+ * https://github.com/dblock/oshi/graphs/contributors
+ */
+package oshi.software.os.mac;
+
+import java.util.List;
+
+import oshi.software.common.AbstractNetworkParams;
+import oshi.util.ExecutingCommand;
+
+public class MacNetworkParams extends AbstractNetworkParams{
+    private final String IPV6_ROUTE_HEADER = "Internet6:";
+
+    private final String DEFAULT_GATEWAY = "default";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDomainName() {
+        return "";
```

Comment:
This overrides the default methods we put in the abstract class.  If you get the correct `LibC`/`SystemB` class to create an instance of in that class, you can just leave this blank and OS X will use the C functions.